flask/app/analyses.py

```diff
@@ -178,6 +178,38 @@ def create_analysis():
     except:  # LOGIN DISABLED
         requester_id = updated_by_id = 1
 
+    if app.config.get("LOGIN_DISABLED") or current_user.is_admin:
+        user_id = request.args.get("user")
+    else:
+        user_id = current_user.user_id
+
+    query = (
+        db.session.query(models.Analysis)
+        .options(contains_eager(models.Analysis.datasets))
+        .join(
+            models.datasets_analyses_table,
+            models.Analysis.analysis_id
+            == models.datasets_analyses_table.columns.analysis_id,
+        )
+        .join(models.Dataset)
+        .join(
+            models.groups_datasets_table,
+            models.Dataset.dataset_id
+            == models.groups_datasets_table.columns.dataset_id,
+        )
+        .join(
+            models.users_groups_table,
+            models.groups_datasets_table.columns.group_id
+            == models.users_groups_table.columns.group_id,
+        )
+        .filter(models.users_groups_table.columns.user_id == user_id)
+    )
```

Comment:
This function is `create_analysis` so the Analysis doesn't exist yet! This means you want to query against Dataset rather than Analysis.